server.elm

```diff
@@ -1559,6 +1849,14 @@ editionscript = """
           }
           // Open link.
         }
+        if(e.which == 90 && (e.ctrlKey || e.metaKey)) {
+          e.preventDefault();
+          if(!undo()) alert("Nothing to undo!");
```

Comment:
Instead of using alert (which is intrusive), please define a function:

    function popupMessage(m) {
      console.log(m);
    }

and use this function `popupMessage` instead of alert. I'll work on nice temporary popup messages for save actions later, so it's better to do it in one place.